framework/web_app/src/boot/stores.ts

```diff
@@ -1,14 +1,25 @@
+import HealthApiService from 'src/services/api/healthApiService'
 import { useMerchantStore } from '../stores/MerchantStore'
 import { useProductStore } from '../stores/ProductStore'
 import { useStockItemStore } from '../stores/stockItemStore'
 import { useStockLevelStore } from '../stores/stockLevelStore'
 
-const merchantStore = useMerchantStore()
-const productStore = useProductStore()
-const stockItemStore = useStockItemStore()
-const stockLevelStore = useStockLevelStore()
+async function waitForApiStartupAsync() {
```

Comment:
![image](https://github.com/BenTalese/DiscountDora/assets/95552720/6f4b989f-79e0-48e9-a0bb-33057df4a146)

Potential issues gpt brought up.
Thoughts?

Reply:
was aware it'd be in an infinite loop which is the issue i raised with you (the forever blank screen), wasn't sure what to do with it, but I guess we could put a timeout as it suggested? i was also trying to put a loading spinner but couldn't figure out how to display a loading spinner before the boot files had finished (there's an order on the documentation)

for store initialisation, thought i remembered at least one was dependent on another but seems that is not the case. we should have at least one point in the file where it waits though because otherwise the stores will still be initialising and the front end will have continued on to render before they're done

as for the error handling, the stores are proooobably handled by the global error handler, but it's impossible for the API to give us an error because both API calls are being caught in the health API service

Reply:
1. Ye not fussed, in the future if we really want we can do max retries & a nice visual server error if that fails

2. You could do something like a Task WaitAll done at the end but the store dependency is a good point. Pinia supports stores in one another so something we need to be aware if we make the initialisation async. Regardless think it is fine atm, we can just be aware in case start up performance becomes an issue.

3. If not we can do something manually to present internal server error on initialisation issues. Atm seems fine as is.